addons/nuxeo-binary-metadata/src/main/java/org/nuxeo/binary/metadata/internals/BinaryMetadataServiceImpl.java

```diff
@@ -252,16 +253,18 @@ public void handleUpdate(List<MetadataMappingDescriptor> mappingDescriptors, Doc
             if (blob != null) {
                 boolean isDirtyMapping = isDirtyMapping(mappingDescriptor, doc);
                 if (isDirtyMapping) {
-                    BlobManager blobManager = Framework.getService(BlobManager.class);
-                    BlobProvider blobProvider = blobManager.getBlobProvider(blob);
-                    // do not write metadata in blobs backed by extended blob providers (ex: Google Drive) or blobs from
-                    // providers that prevent user updates
-                    if (blobProvider != null && (!blobProvider.supportsUserUpdate() || blobProvider.getBinaryManager() == null)) {
-                        return;
+                    if(!mappingDescriptor.isReadOnly()) {
+                	    BlobManager blobManager = Framework.getService(BlobManager.class);
```

Comment:
Indent is wrong because of a tab here. Configure your editor to never use tabes.